packages/embark/src/lib/modules/solidity/index.js

```diff
@@ -129,14 +129,21 @@ class Solidity {
 
         for (let contractFile in json) {
           for (let contractName in json[contractFile]) {
+            let className = contractName;
+            let realContractFile = contractFile;
+            // This is for solc 0.4.18 which outputs weirdly
+            if (className.indexOf(':') > -1) {
```

Comment:
Looks good, nice work.

Would it be possible to extract L135-L139 / L173-L177 into a helper function? Maybe one that returns an object that can then be destructured with `({className, realContractFile}) = someHelper(...);`.